generator/lookml.py

```diff
@@ -29,82 +29,85 @@
 """
 
 
-def _generate_views(
+def _generate_view(
     out_dir: Path,
-    views: Iterable[View],
+    view: View,
     v1_name: Optional[str],
     dryrun,
-) -> Iterable[Path]:
-    for view in views:
-        logging.info(
-            f"Generating lookml for view {view.name} in {view.namespace} of type {view.view_type}"
-        )
-        path = out_dir / f"{view.name}.view.lkml"
-        try:
-            lookml = view.to_lookml(v1_name, dryrun)
-            if lookml == {}:
-                continue
-
-            # lkml.dump may return None, in which case write an empty file
-            path.write_text(FILE_HEADER + (lkml.dump(lookml) or ""))
-            yield path
-        except DryRunError as e:
-            if e.error == Errors.PERMISSION_DENIED and e.use_cloud_function:
-                print(
-                    f"Permission error dry running {view.name}. Copy existing {path} file from looker-hub."
-                )
-                try:
-                    get_file_from_looker_hub(path)
-                    yield path
-                except Exception as ex:
-                    print(f"Skip generating view for {path}: {ex}")
-            else:
-                raise
+) -> Optional[Path]:
+    logging.info(
+        f"Generating lookml for view {view.name} in {view.namespace} of type {view.view_type}"
+    )
+    path = out_dir / f"{view.name}.view.lkml"
+
+    try:
+        lookml = view.to_lookml(v1_name, dryrun)
+        if lookml == {}:
+            return None
+
+        # lkml.dump may return None, in which case write an empty file
+        path.write_text(FILE_HEADER + (lkml.dump(lookml) or ""))
+        return path
+    except DryRunError as e:
+        if e.error == Errors.PERMISSION_DENIED and e.use_cloud_function:
+            print(
+                f"Permission error dry running {view.name}. Copy existing {path} file from looker-hub."
+            )
+            try:
+                get_file_from_looker_hub(path)
+                return path
+            except Exception as ex:
+                print(f"Skip generating view for {path}: {ex}")
+                return None
+        else:
+            raise
 
 
-def _generate_explores(
+def _generate_explore(
     out_dir: Path,
     namespace: str,
-    explores: dict,
+    explore_name: str,
+    explore: Any,
```

Comment:
Similar to my suggestion for `_generate_dashboard()`'s `dashboard_info` argument, if this argument were named `explore_info` then in the function body `explore_by_type` could be named `explore` like it was before without conflict.